.github/workflows/build_dependency_image.yml

```diff
@@ -0,0 +1,62 @@
+---
+# This resuable workflow always builds & tags a Docker image for the Nautobot dependencies stage
+# Published as ghcr.io/nautobot/nautobot-dependencies
+#
+# This allows the same steps to be followed in CI on push as well as a manual trigger if so desired
+name: "Build Dependencies Image"
+on:  # yamllint disable
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  container-build:
+    runs-on: "ubuntu-20.04"
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [ "3.7", "3.8", "3.9", "3.10" ]
+    env:
+      INVOKE_NAUTOBOT_PYTHON_VER: "${{ matrix.python-version }}"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v3"
+      - name: "Get Git Branch Name"
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+        id: "gitbranch"
+      - name: "Set up QEMU"
+        uses: "docker/setup-qemu-action@v1"
+      - name: "Set up Docker Buildx"
+        uses: "docker/setup-buildx-action@v1"
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: "Docker Dependenices Metadata"
+        id: "dockerdepmeta"
+        uses: "docker/metadata-action@v3"
+        with:
+          images: "ghcr.io/nautobot/nautobot-dependencies"
+          flavor: |
+            latest=false
+          tags: |
+            type=raw,value=${{ steps.gitbranch.outputs.branch }}-py${{ matrix.python-version }}
+            type=raw,value=${{ steps.gitbranch.outputs.branch }}-{{sha}}-{{date 'X'}}-py${{ matrix.python-version }}
+            type=raw,value=${{ steps.gitbranch.outputs.branch }},enable=${{ matrix.python-version == 3.7 }}
+            type=raw,value=${{ steps.gitbranch.outputs.branch }}-{{sha}}-{{date 'X'}},enable=${{ matrix.python-version == 3.7 }}
```

Comment:
Do we particularly need the `sha` tags for this image? Or the non-versioned tag for python == 3.7?

Reply:
Done